lib/tasks/valid_test_data_generators/trn_generator.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+ALL_TRNS = (1111..9_999_999) unless defined?(ALL_TRNS)
+
+class TRNGenerator
```

Comment:
Could we not just get away with using Faker instead of rolling our own service? its unlikely we would hit the same TRN for the seed data using this:

```
Faker::Number.unique.number(digits: 7)
```

And if we did it would raise an error in Faker instead of some random spec failure (similar to your 'TRN available list exhausted' error):

https://github.com/faker-ruby/faker/blob/main/lib/helpers/unique_generator.rb#L22

Reply:
that's a fair point, happy to only use faker unless we hit issues later. Maybe we use find or create in the service for that?